system_setup/server/server.py

```diff
@@ -88,3 +92,13 @@ def make_model(self):
     async def wait_for_cloudinit(self):
         self.cloud_init_ok = True
         return
+
+    async def start_site(self, runner: web.AppRunner):
+        port = self.opts.tcp_port
+        if port is None:
+            return await super().start_site(runner)
```

Comment:
Shouldn’t we still call parent start_site() in case we pass a tcp port?

Reply:
`super().start_site()` handles the default behavior (creating a Unix socket site and changing the socket permissions). When variants override this method to create other types of sites, its their responsibility to start it. Internally, SubiquityServer will await this method in its `start_api_server()` method, as you can see here:

https://github.com/canonical/subiquity/blob/a9ac5a5fb28168aa328e2f9453b1f4768fd4e54c/subiquity/server/server.py#L494-L505

This is a little different from the original intent (which was make a site factory), it changed after some conversation I had with Dan Bungert. The spec contains the updated proposal, which was already merged in Subiquity.

Reply:
Hi @didrocks, yes this implementation is consistent with my request to Carlos to ensure that the unix socket creation happens in the same place as the chmod of that socket, which was implemented in #1305.  An earlier plan had them split and the `chmod` was behind a new `if` check, and I didn't want even a little risk that that check could fail to run `chmod`, as Subiquity for Server would be fine but ubuntu-desktop-install would fail.

So +1 to having system-setup override the `start_site()` behavior.